app/airq/sync/purpleair.py

```diff
@@ -127,13 +116,17 @@ def _sensors_sync(
     return moved_sensor_ids
 
 
-def _relations_sync(moved_sensor_ids: typing.List[int], relations_map: TRelationsMap):
+def _relations_sync(moved_sensor_ids: typing.List[int]):
     trie: Trie[Zipcode] = Trie()
     for zipcode in Zipcode.query.all():
         trie.insert(zipcode.geohash, zipcode)
 
     new_relations = []
-    updates = []
+
+    # Delete the old relations before rebuilding them
+    SensorZipcodeRelation.query.filter(
+        SensorZipcodeRelation.sensor_id.in_(moved_sensor_ids)
+    ).delete()
```

Comment:
Can you confirm this really does the deletion? I think you might need to call `db.session.commit()` after.

Reply:
``db.session.commit()`` is needed. Adding...